subprojects/core/src/main/java/org/gradle/api/internal/project/CrossProjectConfigurator.java

```diff
@@ -18,8 +18,9 @@
 
 import org.gradle.api.Action;
 import org.gradle.api.Project;
+import org.gradle.api.internal.WithMutationGuard;
 
-public interface CrossProjectConfigurator {
+public interface CrossProjectConfigurator extends WithMutationGuard {
```

Comment:
❓  Now that we removed the mutation guard from this class, would it make sense to move the mutation guard somewhere else that is instantiated once per build session?